File: node/examples/03-pk-mvp-tracker.js
// Print PK MVPs + booster cards as they fire.
// Run: USERNAME=streamername node 03-pk-mvp-tracker.js
import { TikTokLive } from 'tiktok-live-events';

const live = new TikTokLive(process.env.USERNAME || 'tiktokuser');

live.on('battle', e => console.log(`[battle] status=${e.status} id=${e.battleId} duration=${e.battleDuration}s`));

live.on('battleArmies', e => {
    console.log(`[armies] remaining=${e.secsRemaining}s`);
    for (const host of e.hosts ?? []) {
        console.log(`  Host ${host.hostUserId} total=${host.teamTotalScore}`);
        const mvp = host.contributors?.[0];
        if (mvp) console.log(`    MVP ${mvp.nickname} ${mvp.score} diamonds`);
    }
});

live.on('battleItemCard', e => {
    if (e.multiplier > 0) console.log(`[card] x${e.multiplier} booster from ${e.senderNickname}`);
    else console.log(`[card] effect=${e.effect} from ${e.senderNickname} duration=${e.durationSec}s`);
});

await live.connect();

File: python/examples/03_pk_mvp_tracker.py
"""Print PK MVPs + booster cards as they fire."""
import asyncio
import os
from tiktok_live_events import TikTokLive

live = TikTokLive(os.environ.get("USERNAME", "tiktokuser"))


@live.on("battle")
def on_battle(e):
    print(f"[battle] status={e['status']} id={e['battleId']} duration={e['battleDuration']}s")


@live.on("battleArmies")
def on_armies(e):
    print(f"[armies] remaining={e.get('secsRemaining')}s")
    for host in e.get("hosts", []):
        print(f"  Host {host['hostUserId']} total={host['teamTotalScore']}")
        mvp_list = host.get("contributors", [])
        if mvp_list:
            mvp = mvp_list[0]
            print(f"    MVP {mvp['nickname']} {mvp['score']} diamonds")


@live.on("battleItemCard")
def on_card(e):
    if e.get("multiplier", 0) > 0:
        print(f"[card] x{e['multiplier']} booster from {e['senderNickname']}")
    else:
        print(f"[card] effect={e['effect']} from {e['senderNickname']} duration={e['durationSec']}s")


asyncio.run(live.run())
